tests/backends/storage/test_swift.py

```diff
@@ -0,0 +1,291 @@
+        return {"success": True}
+
+    monkeypatch.setattr(SwiftService, "list", mock_list_with_pages)
+    monkeypatch.setattr(SwiftService, "stat", mock_successful_stat)
+    fs.create_file(HISTORY_FILE, contents=json.dumps(history))
+    swift = swift()
+    assert list(swift.list()) == [x["name"] for x in listing] * pages_count
+    assert list(swift.list(new=True)) == ["2020-05-01.gz"] * pages_count
+    assert list(swift.list(details=True)) == listing * pages_count
+
+
+@pytest.mark.parametrize("pages_count", [1, 2])
+def test_swift_list_with_failed_connection_should_log_the_error(
+    pages_count, swift, monkeypatch, fs, caplog
+):  # pylint:disable=invalid-name
+    """Given SwiftService.list method fails to retrieve the list of archives
+    The SwiftStorage list method should log the error and raise a BackenException
+    """
+
+    def mock_list_with_pages(*args, **kwargs):  # pylint:disable=unused-argument
+        return [
+            {
+                "success": False,
+                "container": "ralph_logs_container",
+                "error": "Container not found",
+            }
+        ] * pages_count
+
+    def mock_successful_stat(*args, **kwargs):  # pylint:disable=unused-argument
+        return {"success": True}
+
+    monkeypatch.setattr(SwiftService, "list", mock_list_with_pages)
+    monkeypatch.setattr(SwiftService, "stat", mock_successful_stat)
+    fs.create_file(HISTORY_FILE, contents=json.dumps([]))
+    caplog.set_level(logging.ERROR)
+    swift = swift()
+    msg = "Failed to list container ralph_logs_container: Container not found"
+    with pytest.raises(BackendException, match=msg):
+        assert list(swift.list()) == []
+    with pytest.raises(BackendException, match=msg):
+        assert list(swift.list(new=True)) == []
+    with pytest.raises(BackendException, match=msg):
+        assert list(swift.list(details=True)) == []
+    logger_name = "ralph.backends.storage.swift"
+    assert caplog.record_tuples == [(logger_name, logging.ERROR, msg)] * 3
+
+
+def test_swift_read_with_valid_name_should_write_to_history(
+    swift, monkeypatch, fs
+):  # pylint:disable=invalid-name
+    """Given SwiftService.download method successfully retrieves from the Swift storage the object
+    with the provided name (the object exists)
+    The SwiftStorage read method should write the entry to the history
+    """
+
+    def mock_successful_download(*args, **kwargs):  # pylint:disable=unused-argument
+        yield {"contents": [b"some", b"contents"]}
+
+    def mock_successful_stat(*args, **kwargs):  # pylint:disable=unused-argument
+        return {"success": True}
+
+    freezed_now = datetime.datetime.now(tz=datetime.timezone.utc).isoformat()
+    monkeypatch.setattr(SwiftService, "download", mock_successful_download)
+    monkeypatch.setattr(SwiftService, "stat", mock_successful_stat)
+    monkeypatch.setattr(swift_module, "now", lambda: freezed_now)
```

Comment:
Why not patching `ralph.utils` instead?

```suggestion
    monkeypatch.setattr(ralph.utils, "now", lambda: freezed_now)
```

Reply:
This makes sense, unfortunately in the `swift.py` I use `from ralph.utils import now` and it seems that this gives the now function a different name. https://stackoverflow.com/questions/31306080/pytest-monkeypatch-isnt-working-on-imported-function

Do you prefer that we change the import to `import ralph.utils as utils` in swift.py or do we change the test monkeypatch ?:
from this:
`monkeypatch.setattr(swift_module, "now", lambda: freezed_now)`
to this:
`monkeypatch.setattr("ralph.backends.storage.swift.now", lambda: freezed_now)`

Reply:
I prefer the later (that you've implemented).